src/modules/peek/Peek.FilePreviewer/Previewers/ImagePreviewer/Helpers/ThumbnailHelper.cs

```diff
@@ -59,5 +61,21 @@ public static HResult GetThumbnail(string filename, out IntPtr hbitmap, NativeSi
 
             return hr;
         }
+
+        public static async Task<BitmapImage> GetThumbnailAsync(string path, uint size)
+        {
+            // preview image
+            var file = await Windows.Storage.StorageFile.GetFileFromPathAsync(path);
+
+            var imageStream = await file.GetThumbnailAsync(
```

Comment:
How is the performance on this? If I recall correctly, the first iteration of the migration used the Windows Storage API and it was much slower than what we have currently